File: cpp/Faculty.cpp
#include "Faculty.h"
#include <iostream>

Faculty::Faculty(int id, const std::string& name, const std::string& email)
    : id(id), name(name), email(email) {}

void Faculty::teachCourse(const Course& course) const {
    std::cout << name << " teaches " << course.name << std::endl;
}

File: csharp/Faculty.cs
using System;

namespace StudentManagementSystem
{
    public class Faculty
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }

        public Faculty(int id, string name, string email)
        {
            Id = id;
            Name = name;
            Email = email;
        }

        public void TeachCourse(Course course)
        {
            Console.WriteLine($"{Name} teaches {course.Name}");
        }
    }
}
